src/folderlistmodel.cpp

```diff
@@ -402,6 +487,38 @@ void FolderListModel::updateCurrentFolderIndex()
     setCurrentFolderIdx(0);
 }
 
+int FolderListModel::addFolderAndChildren(const QMailFolderId &folderId, QMailMessageKey messageKey,
+                                          QList<QMailFolderId> &originalList, int indexIterator)
+{
+    for (int i = 0; i < originalList.size(); i++) {
```

Comment:
Instead of this outer loop: "int i = originalList.indexOf(folderId); if (i != -1) { ... }"